app/controllers/bookings_controller.rb

```diff
@@ -52,14 +52,33 @@ def scan_ticket
 
   def request_refund
     @booking = Booking.find_by_uniq_id(params[:uniq_id])
-    flash[:notice] = if @booking.update_attributes(
-      refund_requested: true,
-      time_requested: Time.now
-    )
-                       "Request for a refund has been sent"
+    unless @booking.refund_requested
+      @booking.update_attributes(
+        refund_requested: true,
+        time_requested: Time.now,
+        reason: params[:reason]
+      )
+    end
+  end
+
+  def grant_refund
+    @booking = Booking.find_by_uniq_id(params[:uniq_id])
```

Comment:
**Nitpick - Code Quality**
You should make use of `find_by` method instead. 

You'd have something like this:
```
Booking.find_by(uniq_id: params[:uniq_id]
```